prow/cmd/gerrit/main.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// package main is a gerrit fetcher updating against a target gerrit instance
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"os/signal"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/gerrit"
+	"k8s.io/test-infra/prow/kube"
+	"k8s.io/test-infra/prow/logrusutil"
+)
+
+var (
+	configPath = flag.String("config-path", "/etc/config/config", "Path to config.yaml.")
+
+	instance = flag.String("gerrit-instance", "", "URL to gerrit instance")
+	projects = flag.String("gerrit-projects", "", "comma separated gerrit projects to fetch from the gerrit instance")
+	storage  = flag.String("storage", "", "Path to persistent volume to load the last sync time")
+)
+
+func main() {
+	flag.Parse()
+	logrus.SetFormatter(
+		logrusutil.NewDefaultFieldsFormatter(nil, logrus.Fields{"component": "gerrit"}),
+	)
+
+	if *instance == "" {
+		logrus.Fatal("--gerrit-intance must set")
+	}
+
+	projs := strings.Split(*projects, ",")
+	if len(projs) == 0 {
+		logrus.Fatal("must have one or more target gerrit project")
+	}
+
+	// authenticate to gerrit server
+	if err := exec.Command("./git-cookie-authdaemon").Run(); err != nil {
```

Comment:
does this need to be added to a docker image?

Reply:
yeah, I haven't written the image rule yet